pkg/internal/computeresources/tasklevel/tasklevel_test.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tasklevel_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/internal/computeresources/tasklevel"
+	"github.com/tektoncd/pipeline/test/diff"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestApplyTaskLevelResourceRequirements(t *testing.T) {
+	testcases := []struct {
+		desc                     string
+		Steps                    []v1beta1.Step
+		ComputeResources         corev1.ResourceRequirements
+		expectedComputeResources []corev1.ResourceRequirements
+	}{{
+		desc: "only with requests",
+		Steps: []v1beta1.Step{{
+			Name:    "1st-step",
+			Image:   "image",
+			Command: []string{"cmd"},
+		}, {
+			Name:    "2nd-step",
+			Image:   "image",
+			Command: []string{"cmd"},
+		}},
+		ComputeResources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+		}, {
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+		}},
+	}, {
+		desc: "only with limits",
+		Steps: []v1beta1.Step{{
+			Name:    "1st-step",
+			Image:   "image",
+			Command: []string{"cmd"},
+		}, {
+			Name:    "2nd-step",
+			Image:   "image",
+			Command: []string{"cmd"},
+		}},
+		ComputeResources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
```

Comment:
Since the limits are "500m", we also want to set the total requests to "500m", meaning that each container should have requests of "250m".

Reply:
I see. this (limits) will be applied as the total task-level allowed requests.
let me correct.

Reply:
corrected by "spreading" the limits when no requests.